### DISCUSSION TITLE
CMake - CS自学指南

### DISCUSSION THREAD

**Discussion by giscus** [PKUFlyingPig/cs-self-learning#101] [Announcements]
# CMake - CS自学指南

CS自学指南

https://csdiy.wiki/%E5%BF%85%E5%AD%A6%E5%B7%A5%E5%85%B7/CMake/

**Comment by duanchang425** [comment]
我很推荐CMake

**Reply by CS-Trekker** [reply]
支持，虽然我不会

**Reply by leohu1** [reply]
cmake比GNU Make用着舒服

**Comment by LHG41278** [comment]
https://llvm.org/docs/CMakePrimer.html#ft-view
如果对如何用cmake去构建最基本的项目这个过程有了一定的了解，可以用这篇llvm社区写的cmake文档来强化对cmake里面的基本语法的理解，这个比cmake官方文档的可读性要强得多

**Comment by Cpt11** [comment]


**Reply by lismos114** [reply]
谁问你了

**Comment by TephrocactusMYC** [comment]
如果要看书的话，推荐《Modern CMake for C++》，讲解还算清楚，作为入门绝对够用了，另一本工具书是《Professional CMake A Practical Guide》，可以当作字典查阅。

**Reply by NYfLAmE** [reply]
thx

**Reply by ghost** [reply]
对的对的, Modern CMake for C++ 2nd好书, 非常系统的介绍了软件构建过程(software build process), 还提供了对软件构建过程的每个方面提供了深入阅读材料.

**Comment by We-unite** [comment]
初学者求助各位，是否有一些简单明了的、使用cmake的c/cpp项目能够帮助理解cmake，感觉干看知识点有点费人，一下子记不住这老些内容

**Reply by ZuoAoyu** [reply]
CMake 官方 Tutorial 总共分为12个步骤（step），跟着一点点把这个项目做完，期间随时查看cmake的官方文档，之后就对cmake有一个“高屋建瓴”的理解了。

**Comment by NGUAlexZhang** [comment]
有一说一，我觉得cmake比makefile舒服啊，因为不用写那么冗长的gcc命令

**Comment by Mysvac** [comment]
CMake本质上算是一门小语言，支持变量、数组、宏、函数、循环与分支，所以应该把它当做一门语言来快速入门。
推荐B站“原子之音”的《现代C++: CMake简明教程》，一个晚上就足够掌握基础了。

**Comment by lil-Journey** [comment]
自建日本节点服务器 月租5元不限流量 本人也在用 失效时可联系人工修复 qq:1165116287

**Comment by songbaii** [comment]
https://cmake.com.cn/cmake/help/latest/guide/tutorial/index.html，这个网站对cmake的教程进行了翻译，比用浏览器自带的翻译看英文教程要舒服得多。

**Reply by songbaii** [reply]
https://cmake.com.cn/cmake/help/latest/guide/tutorial/index.html
链接是这个，上面不知道为什么网址和评论黏在一起了。
